src/Castle.Core/DynamicProxy/ProxyUtil.cs

```diff
@@ -80,5 +82,36 @@ public static bool IsProxyType(Type type)
 		{
 			return typeof(IProxyTargetAccessor).IsAssignableFrom(type);
 		}
+
+		/// <summary>
+		/// Checks to see if a method is accessible to DynamicProxyGenAssembly2.</summary>
+		/// <param name="method">The method to check</param>
+		/// <param name="message">If the method is accessible, <c>null</c>; otherwise, an explanation of why the method is not accessible</param>
+		/// <returns><c>true</c> if the method is accessible, <c>false</c> otherwise</returns>
+		public static bool IsAccessible(MethodBase method, out string message)
```

Comment:
While we are here I'm thinking it would be worthwhile adding a `public static bool IsAccessible(MethodBase method)` (non-extension method) overload that just calls the InternalsUtil extension method.